### PR TITLE
feat: add native msfs2024 support

### PR DESCRIPTION
**Pull request by tzobler**
"Microsoft flight simulator 2024" has been integrated into all functions. The datastore also contains information about this simulator

### CODE REVIEW

File: src/misc/simulation/fscommon/fsdirectories.cpp [hunk 1]
```diff
@@ -263,6 +336,35 @@ namespace swift::misc::simulation::fscommon
...
+            "OneStore/asobo-vcockpits",
+            "OneStore/asobo-l",
+            "OneStore/asobo-m",
+            "OneStore/asobo-vfx",
+            "OneStore/fs",
+            "OneStore/esd",
+            "OneStore/microsoft-airport",
+            "OneStore/microsoft-bushtrip",
+            "OneStore/microsoft-discovery",
+            */
```
Comment (ltoenning) [inline src/misc/simulation/fscommon/fsdirectories.cpp:360]: Remove if they are no longer required.

File: src/plugins/simulator/CMakeLists.txt [hunk 2]
```diff
@@ -37,10 +37,13 @@ endif()
...
+
+if(SWIFT_BUILD_MSFS2024_PLUGIN)
+    add_subdirectory(msfs2024)
 endif()
 
 if(SWIFT_BUILD_FSX_PLUGIN)
     add_subdirectory(fsx)
     add_subdirectory(fsxconfig)
-endif()
+endif()
```
Comment (ltoenning) [inline src/plugins/simulator/CMakeLists.txt:49]: Here a newline (\n) at the end of the file is missing

File: src/plugins/simulator/fscommon/fsuipcimpl.cpp [hunk 3]
```diff
@@ -296,6 +296,7 @@ namespace swift::simplugin::fscommon
                 pitchRaw = qRound(std::floor(pitchRaw * angleCorrectionFactor));
                 bankRaw = qRound(std::floor(bankRaw * angleCorrectionFactor));
 
+                // TODO TZ: should MSFS2024 use FSUIPC?
```
Comment (ltoenning) [inline src/plugins/simulator/fscommon/fsuipcimpl.cpp:299]: What are the benefits of using FSUIPC?

File: src/plugins/simulator/fsxcommon/simconnectdatadefinition.cpp [hunk 4]
```diff
@@ -424,6 +425,23 @@ namespace swift::simplugin::fsxcommon
         return hr;
     }
 
+    // TODO TZ: are there any changes in MSFS2024 ?
```
Comment (ltoenning) [inline src/plugins/simulator/fsxcommon/simconnectdatadefinition.cpp:428]: At least from the docs, nothing has changed:

- https://docs.flightsimulator.com/msfs2024/html/6_Programming_APIs/SimVars/Aircraft_SimVars/Aircraft_RadioNavigation_Variables.htm#TRANSPONDER_STATE
- https://docs.flightsimulator.com/msfs2024/html/6_Programming_APIs/SimVars/Aircraft_SimVars/Aircraft_RadioNavigation_Variables.htm#TRANSPONDER_IDENT

File: CMakeLists.txt [hunk 5]
```diff
@@ -32,8 +32,8 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/out/bin)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/out/lib)
 
 # Build related options
-option(SWIFT_BUILD_SAMPLES "Build samples" ON)
-option(SWIFT_BUILD_UNIT_TESTS "Build unit tests" ON)
+option(SWIFT_BUILD_SAMPLES "Build samples" OFF)
+option(SWIFT_BUILD_UNIT_TESTS "Build unit tests" OFF)
```
[the remark this answers is not in the document]
Comment (tzobler) [reply CMakeLists.txt:36]: that was for my local performance. restored

File: cmake/tools.cmake [hunk 6]
```diff
@@ -29,11 +29,11 @@ macro(load_vatsim_key)
...
-    if(VATSIM_KEY_JSON)
-        file(READ ${PROJECT_SOURCE_DIR}/${VATSIM_KEY_JSON} KEY_FILE)
-        string(JSON SWIFT_VATSIM_ID GET ${KEY_FILE} vatsim id)
-        string(JSON SWIFT_VATSIM_KEY GET ${KEY_FILE} vatsim key)
-    endif()
+#    if(VATSIM_KEY_JSON)
+#        file(READ ${PROJECT_SOURCE_DIR}/${VATSIM_KEY_JSON} KEY_FILE)
+#        string(JSON SWIFT_VATSIM_ID GET ${KEY_FILE} vatsim id)
+#        string(JSON SWIFT_VATSIM_KEY GET ${KEY_FILE} vatsim key)
+#    endif()
```
[the remark this answers is not in the document]
Comment (tzobler) [reply cmake/tools.cmake:36]: restored

File: src/gui/components/dbownmodelscomponent.cpp [hunk 7]
```diff
@@ -393,15 +393,29 @@ namespace swift::gui::components
...
+                if (!m_loadActions[6])
+                {
+                    m_loadActions[6] = new QAction(CIcons::appModels16(), "MSFS2024 models", this);
+                    connect(m_loadActions[6], &QAction::triggered, ownModelsComp, [ownModelsComp](bool checked) {
+                        if (!ownModelsComp) { return; }
+                        Q_UNUSED(checked)
+                        ownModelsComp->setSimulator(CSimulatorInfo::msfs2024(), true);
+                    });
+                }
+                menuActions.addAction(m_loadActions[5], CMenuAction::pathSimulator());
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/gui/components/dbownmodelscomponent.cpp:407]: changed

File: src/misc/simulation/fscommon/fsdirectories.cpp (hunk 1 above)
Comment (tzobler) [reply src/misc/simulation/fscommon/fsdirectories.cpp:360]: will probably be valid again when Asobo activates the marketplace. Deleted for now. Template is available

File: src/plugins/simulator/fsxcommon/simconnectdatadefinition.cpp (hunk 4 above)
Comment (tzobler) [reply src/plugins/simulator/fsxcommon/simconnectdatadefinition.cpp:428]: Comment removed

File: src/plugins/simulator/fscommon/fsuipcimpl.cpp (hunk 3 above)
Comment (tzobler) [reply src/plugins/simulator/fscommon/fsuipcimpl.cpp:299]: inactive code. Comment removed

File: src/plugins/simulator/CMakeLists.txt (hunk 2 above)
Comment (tzobler) [reply src/plugins/simulator/CMakeLists.txt:49]: added. hopefully correct

**Pull request closed without merging**